presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestRemoveIdentityCasts.java

```diff
@@ -0,0 +1,127 @@
+import com.facebook.presto.tpch.TpchConnectorFactory;
+import com.google.common.collect.ImmutableMap;
+import org.intellij.lang.annotations.Language;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.anyTree;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.project;
+import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
+
+public class TestRemoveIdentityCasts
+{
+    private final LocalQueryRunner queryRunner;
+
+    public TestRemoveIdentityCasts()
+    {
+        this.queryRunner = new LocalQueryRunner(testSessionBuilder()
+                .setCatalog("local")
+                .setSchema("tiny")
+                .build());
+
+        queryRunner.createCatalog(queryRunner.getDefaultSession().getCatalog().get(),
+                new TpchConnectorFactory(queryRunner.getNodeManager(), 1),
+                ImmutableMap.<String, String>of());
+    }
+
+    @Test
+    public void testRemoveIdentityCastUnparameterizedColumnData()
+    {
+        @Language("SQL") String sql = "SELECT CAST(nationkey AS bigint) FROM nation";
+        PlanMatchPattern pattern = anyTree(
+                project(anyTree()).withAssignment("Cast")
+        );
+        assertNotPlan(sql, pattern);
+    }
+
+    @Test
+    public void testKeepNonIdentityCastUnparameterizedColumnData()
+    {
+        @Language("SQL") String sql = "SELECT CAST(nationkey AS smallint) FROM nation";
+        PlanMatchPattern pattern = anyTree(
+                project(anyTree()).withAssignment("Cast")
+        );
+        assertPlan(sql, pattern);
+    }
+
+    @Test
+    public void testRemoveIdentityCastParameterizedColumnData()
+    {
+        @Language("SQL") String sql = "SELECT CAST(name AS varchar(25)) FROM nation";
+        PlanMatchPattern pattern = anyTree(
+                project(anyTree()).withAssignment("Cast")
+        );
+        assertNotPlan(sql, pattern);
+    }
+
+    @Test
+    public void testKeepNonIdentityCastParameterizedColumnData()
+    {
+        @Language("SQL") String sql = "SELECT CAST(name AS varchar(30)) FROM nation";
+        PlanMatchPattern pattern = anyTree(
+                project(anyTree()).withAssignment("Cast")
+        );
+        assertPlan(sql, pattern);
+    }
+
+    @Test
+    public void testKeepNonIdentityCastDefaultParameterizedColumnData()
+    {
+        @Language("SQL") String sql = "SELECT CAST(name AS varchar) FROM nation";
+        PlanMatchPattern pattern = anyTree(
+                project(anyTree()).withAssignment("Cast")
+        );
+        assertPlan(sql, pattern);
+    }
+
+    @Test
+    public void testRemoveIdentityCastImplicitBigintData()
+    {
+        @Language("SQL") String sql = "SELECT CAST(bigint '5' AS bigint)";
+        PlanMatchPattern pattern = anyTree(
+                project(anyTree()).withAssignment("Cast")
+        );
+        assertNotPlan(sql, pattern);
```

Comment:
Frankly, I don't like `assertNotPlan` idea because what you check here does not guarantee that plan is still good even though it does not have casting.
Maybe it would be possible to do something like:
```java
assertPlan(sql, anyNot(project().withAssignment("Cast")));
```

Obviously this is just a gist of an idea because current implementation of `anyNot` simply compares classes.

If it's too hard to change it, let's leave it as is.